runner/query_run.go

```diff
@@ -39,28 +39,63 @@ func SetupQueryRunnerForAllServers(ctx context.Context, servers []*state.Server,
 	}
 }
 
+func runQueryOnDatabase(ctx context.Context, server *state.Server, collectionOpts state.CollectionOpts, logger *util.Logger, id int64, query *state.QueryRun) (string, error) {
+	db, err := postgres.EstablishConnection(ctx, server, logger, collectionOpts, query.DatabaseName)
+	if err != nil {
+		return "", err
+	}
+	defer db.Close()
+
+	if postgres.StatsHelperExists(ctx, db, "explain_analyze") {
```

Comment:
This `runQueryOnDatabase` function is expected to be used for other types of QueryRun too, but this helper method check, as well as Postgres settings applying queries are used for any types of QueryRun.
I wonder if we should actually check the type in the beginning of this func and early return unless `query.Type == pganalyze_collector.QueryRunType_EXPLAIN` 🤔